src/mds/journal.cc

```diff
@@ -2496,6 +2494,12 @@ void ESlaveUpdate::generate_test_instances(std::list<ESlaveUpdate*>& ls)
   ls.push_back(new ESlaveUpdate());
 }
 
+void ESlaveUpdate::update_segment() {
+    if (op == ESlaveUpdate::OP_PREPARE) {
+      auto segment = get_segment();
+      segment->uncommitted_slaves.insert(reqid);
```

Comment:
this code should be moved into MDCache::add_uncommitted_slave

Reply:
Done.